apps/server/src/connect/dataEndpoints.ts

```diff
@@ -4,52 +4,37 @@ import Joi from 'joi'
 import { getProviderAdapter } from '../adapters'
 import getVC from '../services/vcProviders'
 import { Providers } from '../shared/contract'
-import { createProviderValidator } from '../utils/validators'
 
 export interface AccountsDataQueryParameters {
-  connection_id: string
+  connectionId: string
   provider: string
-  user_id: string
+  userId: string
 }
 
 export interface AccountsRequest {
-  query: AccountsDataQueryParameters
+  params: AccountsDataQueryParameters
 }
 
 export interface IdentityRequest {
-  query: IdentityDataQueryParameters
+  params: IdentityDataParameters
 }
 
 export interface TransactionsRequest {
   query: TransactionsDataQueryParameters
+  params: TransactionsDataPathParameters
 }
 
 export const accountsDataHandler = async (
   req: AccountsRequest,
   res: Response
 ) => {
-  const schema = Joi.object({
-    connection_id: Joi.string().required(),
-    provider: createProviderValidator(),
-    user_id: Joi.string().required()
-  })
-
-  const { error } = schema.validate(req.query)
-
-  if (error) {
-    res.status(400)
-    res.send(error.details[0].message)
-
-    return
-  }
-
-  const { provider, connection_id, user_id } = req.query
+  const { provider, connectionId, userId } = req.params
 
   const providerAdapter = getProviderAdapter(provider)
-  const providerUserId = await providerAdapter.ResolveUserId(user_id)
```

Comment:
Since you took out provider validation I think this line will cause a 500 error when the provider isn't valid. 

You could move these methods into the "try" clause and then uncomment [this error](https://github.com/Universal-Connect-Project/ucw-app/blob/57e083c465d0c0db0fe7150e91ca7ff8c52c0588/apps/server/src/adapters/index.ts#L42), then on the response you'd probably want to send the error message.

Reply:
I like that idea